cmd/gen-key/ecdsa.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/miekg/pkcs11"
+)
+
+var stringToCurve = map[string]*elliptic.CurveParams{
+	elliptic.P224().Params().Name: elliptic.P224().Params(),
+	elliptic.P256().Params().Name: elliptic.P256().Params(),
+	elliptic.P384().Params().Name: elliptic.P384().Params(),
+	elliptic.P521().Params().Name: elliptic.P521().Params(),
+}
+
+var curveToOIDDER = map[*elliptic.CurveParams][]byte{
+	elliptic.P224().Params(): []byte{6, 5, 43, 129, 4, 0, 33},
```

Comment:
You can't really use Params as a map key because it contains `*big.Int`, which will be compared based on their pointer values rather than their integer values. Same below.

Reply:
This works because `.Params()` returns a static struct that is initialized at import time by `elliptic`, essentially it's the exact same thing that is returned by `.Curve()` but not wrapped in a interface. Again I think this should be perfectly fine since making these return value non-comparable (i.e. by returning a fresh struct for each call) would be a breaking API change.

Reply:
It would break an implicit API, not an explicit one, and you can only tell it's comparable by reading the source. Is there a reason not to index by name here also?